packages/core/src/fhir-deduplication/resources/procedure.ts

```diff
@@ -93,18 +100,24 @@ export function groupSameProcedures(procedures: Procedure[]): {
   }
 
   for (const procedure of procedures) {
-    const date = getPerformedDateFromResource(procedure, "datetime");
-    if (!date) continue;
+    if (hasBlacklistedText(procedure.code)) {
+      danglingReferencesSet.add(createRef(procedure));
+      continue;
+    }
+
+    const datetime = getPerformedDateFromResource(procedure, "datetime");
+    if (!datetime) {
+      danglingReferencesSet.add(createRef(procedure));
+      continue;
+    }
 
     const { cptCode, loincCode, snomedCode } = extractCodes(procedure.code);
 
-    const key = cptCode
-      ? JSON.stringify({ date, cptCode })
-      : loincCode
-      ? JSON.stringify({ date, loincCode })
-      : snomedCode
-      ? JSON.stringify({ date, snomedCode })
-      : undefined;
+    let key;
+    if (cptCode) key = JSON.stringify({ datetime, cptCode });
+    else if (loincCode) JSON.stringify({ datetime, loincCode });
```

Comment:
this looks funny. Shouldnt the else ifs have key = ....

Reply:
good catch!

Reply:
fixed